pkg/apis/build/v1alpha1/build_validation.go

```diff
@@ -19,18 +19,43 @@ func (b *Build) Validate(ctx context.Context) *apis.FieldError {
 
 func (bs *BuildSpec) Validate(ctx context.Context) *apis.FieldError {
 	return validateListNotEmpty(bs.Tags, "tags").
+		Also(validateTags(bs.Tags)).
 		Also(bs.Builder.Validate(ctx).ViaField("builder")).
-		Also(bs.Source.Validate(ctx).ViaField("source"))
+		Also(bs.Source.Validate(ctx).ViaField("source")).
+		Also(bs.LastBuild.Validate(ctx).ViaField("lastBuild")).
+		Also(bs.validateImmutableFields(ctx))
 }
 
-func (bbs *BuildBuilderSpec) Validate(ctx context.Context) *apis.FieldError {
-	if bbs.Image == "" {
-		return apis.ErrMissingField("image")
+func (bs *BuildSpec) validateImmutableFields(ctx context.Context) *apis.FieldError {
```

Comment:
Builds only execute once and will not reschedule build pods if any changes are made to the `build.spec`. Therefore, the entire `build.spec` should be immutable to prevent users from updating a resource limit and expecting it to alter the already scheduled build pod.

Reply:
so this will not come into play when the image spec has been updated with updated resource constrains right?

Reply:
Correct, in kpack builds are immutable and kpack always creates new builds instead of modifying old ones when a image spec changes.